src/micro_http/src/server.rs

```diff
@@ -379,6 +379,19 @@ impl HttpServer {
         Ok(parsed_requests)
     }
 
+    /// This function is responsible with flushing any remaining outgoing
+    /// requests on the server.
+    ///
+    /// Note that this function can block the thread on write, since the
+    /// operation is blocking.
+    pub fn flush_outgoing_writes(&mut self) {
+        for (_, connection) in self.connections.iter_mut() {
+            if connection.state == ClientConnectionState::AwaitingOutgoing {
```

Comment:
should this be a loop?
```suggestion
            while connection.state == ClientConnectionState::AwaitingOutgoing {
```

Reply:
Can it be an infinite loop ? What if we cannot send data back to the client ?

Reply:
@acatangiu yes, this could also be a loop. Thumbs up for the suggestion!
@sandreim yes, it can block on write if the operation waits on the client

Reply:
Can we have a timeout for the blocking case ? so that the behavior is predictible.

Reply:
hmm.. I can't think of any non-complex solution to handling the blocking case.

Is the extra complexity necessary though? Isn't it the same behavior as all other API responses?

It is a good catch @sandreim, but **if there is no easy way to do it**, I say take it as is.
I wouldn't invest more time and adding more complexity for these unlikely error paths which are irrelevant in production scenarios, since that comes at the expense of investing in things with much higher impact and value.

Reply:
Every FD backing a stream is set to non-blocking before creating the connection (`server.rs:489`). Writes can't be blocking as we would immediately receive `EAGAIN` or `EWOULDBLOCK`. As long as we check for these two error cases, the loop can't be infinite.

Reply:
`connection.write()`  checks for these cases through a more general StreamError approach, so we are covered in this regard. The respective does close the connection whenever it encounters an unexpected StreamError or a ConnectionClosed error.